### PR TITLE
Add a global illumination demo (4.0-dev)

### PR DESCRIPTION
**Pull request by Calinou**
Port of the 3.x global illumination demo, with SDFGI support added and screen-space lighting effect options added (SSAO, SSIL, or both).

Lightmap options were removed as I couldn't get both indirect-only and fully baked lightmaps to work for now. They can be readded in the future.

See https://github.com/godotengine/godot-demo-projects/issues/697.

### CODE REVIEW

**Comment by Calinou** [comment]
> This PR has conflicts, it needs to be rebased (you can probably rebase everything in favor of your branch). Also, the commit on this PR says it's unverified, did your GPG key expire?

Fixed. I tried to rebase yesterday, but I rebased against `master` instead of `4.0-dev`.

**Review by aaronfranke — APPROVED** [review]

File: 3d/global_illumination/camera.gd [hunk 1]
```diff
@@ -29,9 +29,9 @@ func _input(event):
 
 func _process(delta):
 	var motion = Vector3(
-			Input.get_axis(&"move_left", &"move_right"),
+			Input.get_action_strength("move_right") - Input.get_action_strength("move_left"),
 			0,
-			Input.get_axis(&"move_forward", &"move_back")
+			Input.get_action_strength("move_back") - Input.get_action_strength("move_forward")
```
Comment (aaronfranke) [inline 3d/global_illumination/camera.gd:34]: Wait, why did you change this? `Input.get_axis(&"move_forward", &"move_back")` is what we want.

Comment (Calinou) [reply 3d/global_illumination/camera.gd:34]: I don't remember – it's likely accidental, as I don't remember performing this change.
